### ISSUE TITLE
BatchNorm with Momentum=0.0 Should Return the Same Output with training=True, and then with Training=False

### ISSUE THREAD

**Issue by benjamink85** [tensorflow/tensorflow#41979]
Python 3.6
Ubuntu 18.04
Tensorflow from Binary
v2.3.0-rc2-23-gb36436b087 2.3.0

```
import tensorflow as tf

seq1 = tf.keras.Sequential([tf.keras.layers.Conv2D(32, 3), tf.keras.layers.BatchNormalization(axis=3, momentum=0.0), tf.keras.layers.LeakyReLU(0.01)])

import numpy as np
x = np.random.randn(128,32,32,1)

res1 = seq1(x, training=True)
res2 = seq1(x, training=False)

print(np.linalg.norm(res1 - res2))
```

When using BatchNormalization with momentum=0.0 with training=True, the moving average of mean and variance should update to the statistics of the batch exactly. Therefore when running the model again on the same data with training=False one should expect to get the same output. 
But the print shows a difference between the two of 0.005684062
Help will be appreciated.
Thank you

**Comment by Saduf2019** [comment]
@benjamink85 
This is not any bug related issue or feature request for Tensorflow, hence request you to report this on [StackOverflow](https://stackoverflow.com/questions/tagged//tensorflow) as its a bigger community to help.
